### PR TITLE
Fix #2015. Save server urls in SharedPref only on successful login

### LINKED ISSUE #2015: Logging in uses the custom server details even if the 'advanced options' checkbox is unchecked
Particularly troublesome if you log in on a custom homeserver, log out, then try to log in to matrix.org.

### PR DESCRIPTION
**Pull request by bmarty**
Fix #2015 

+ Do some cleanup

### CODE REVIEW

**Comment by bmarty** [comment]
I've rebased the branch to develop and fix the issue. New SharedPrefs keys has been added to store the default Matrix server urls from the referrer. PR is ready to be reviewed

File: vector/src/main/java/im/vector/activity/util/RequestCodes.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package im.vector.activity.util
```
Comment (manuroe) [inline vector/src/main/java/im/vector/activity/util/RequestCodes.kt]: Copyright is missing

File: vector/src/main/java/im/vector/fragments/VectorBaseFragment.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package im.vector.fragments
```
Comment (manuroe) [inline vector/src/main/java/im/vector/fragments/VectorBaseFragment.kt]: Copyright

File: vector/src/main/java/im/vector/repositories/ServerUrlsRepository.kt [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package im.vector.repositories
```
Comment (manuroe) [inline vector/src/main/java/im/vector/repositories/ServerUrlsRepository.kt]: copyright

File: vector/src/main/java/im/vector/util/PreferencesManager.java [hunk 4]
```diff
@@ -185,9 +185,9 @@
     public static void clearPreferences(Context context) {
         Set<String> keysToKeep = new HashSet<>(mKeysToKeepAfterLogout);
 
-        // home server url
-        keysToKeep.add(LoginActivity.HOME_SERVER_URL_PREF);
```
Comment (manuroe) [inline vector/src/main/java/im/vector/util/PreferencesManager.java]: I cannot see if LoginActivity.HOME_SERVER_URL_PREF has been removed.

Comment (manuroe) [reply vector/src/main/java/im/vector/util/PreferencesManager.java]: I have found it. My bad.

File: vector/src/main/java/im/vector/activity/util/RequestCodes.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/activity/util/RequestCodes.kt]: Copyright added

File: vector/src/main/java/im/vector/fragments/VectorBaseFragment.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/fragments/VectorBaseFragment.kt]: Copyright added

File: vector/src/main/java/im/vector/repositories/ServerUrlsRepository.kt (hunk 3 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/repositories/ServerUrlsRepository.kt]: Copyright added

**Pull request merged**